src/components/refactor/dialogs/regulating-terminal/regulating-terminal-form-utils.js

```diff
@@ -10,36 +10,41 @@ import {
     ID,
     NAME,
     NOMINAL_VOLTAGE,
+    REGULATING_TERMINAL,
     SUBSTATION_ID,
     TOPOLOGY_KIND,
     TYPE,
     VOLTAGE_LEVEL,
 } from 'components/refactor/utils/field-constants';
 import yup from '../../utils/yup-config';
 
-const regulatingTerminalValidationSchema = () => ({
-    [VOLTAGE_LEVEL]: yup
-        .object()
-        .nullable()
-        .shape({
-            [ID]: yup.string(),
-            [NAME]: yup.string(),
-            [SUBSTATION_ID]: yup.string(),
-            [NOMINAL_VOLTAGE]: yup.string(),
-            [TOPOLOGY_KIND]: yup.string().nullable(),
-        }),
-    [EQUIPMENT]: yup
-        .object()
-        .nullable()
-        .shape({
-            [ID]: yup.string(),
-            [NAME]: yup.string().nullable(),
-            [TYPE]: yup.string(),
-        }),
+const regulatingTerminalValidationSchema = (id) => ({
+    [id]: yup.object().shape({
+        [VOLTAGE_LEVEL]: yup
+            .object()
+            .nullable()
+            .required()
+            .shape({
+                [ID]: yup.string(),
+                [NAME]: yup.string(),
+                [SUBSTATION_ID]: yup.string(),
+                [NOMINAL_VOLTAGE]: yup.string(),
+                [TOPOLOGY_KIND]: yup.string().nullable(),
+            }),
+        [EQUIPMENT]: yup
+            .object()
+            .nullable()
+            .required()
+            .shape({
+                [ID]: yup.string(),
+                [NAME]: yup.string().nullable(),
+                [TYPE]: yup.string(),
+            }),
+    }),
 });
 
 export const getRegulatingTerminalValidationSchema = () => {
-    return regulatingTerminalValidationSchema();
+    return regulatingTerminalValidationSchema(REGULATING_TERMINAL);
 };
 
 const regulatingTerminalEmptyFormData = () => ({
```

Comment:
this utility class got very confusing, do we need [id], or not ? 
Also, we are calling "getRegulatingTerminalValidationSchema" without an id, but it adds one for us for "regulatingTerminalValidationSchema".
And I have the feeling we are breaking other components using this utility file